### PR TITLE
Port SBOM generation to v3.x/ps7

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-powershell-worker/issues/719

Cherry-picked https://github.com/Azure/azure-functions-powershell-worker/pull/716, https://github.com/Azure/azure-functions-powershell-worker/pull/718, https://github.com/Azure/azure-functions-powershell-worker/pull/720, https://github.com/Azure/azure-functions-powershell-worker/pull/728, https://github.com/Azure/azure-functions-powershell-worker/pull/731 applicable changes to the `v3.x/ps7` branch.

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by davidmrdavid — APPROVED** [review]
It's a simple cherry pick, so LGTM

File: azure-pipelines.yml
Comment (michaelpeng36) [inline azure-pipelines.yml:18]: In some cases, the VM Image may actually be ignored, so you may need to replace the line after this. See [this](https://github.com/Azure/azure-functions-core-tools/pull/2883/files) PR for reference.

File: azure-pipelines.yml
Comment (michaelpeng36) [inline azure-pipelines.yml]: I think this will work, but in general, `true` and `false` should be strings `'true'` and `'false'`, see the examples in [this doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml)

File: azure-pipelines.yml
Comment (michaelpeng36) [inline azure-pipelines.yml]: The same comment as before: `true` should be `'true'` here.

**Review by michaelpeng36 — CHANGES_REQUESTED** [review]
Some minor changes requested, but the rest looks good to me.

File: azure-pipelines.yml
Comment (Francisco-Gamino) [reply azure-pipelines.yml:18]: As per our conversation, let wait until @eliaslopezgt merge the change into the other repros and then we can make the change here. I have opened issue https://github.com/Azure/azure-functions-powershell-worker/issues/730 to track this.

File: azure-pipelines.yml
Comment (Francisco-Gamino) [reply azure-pipelines.yml]: Fixed in the latest cherry-pick.

File: azure-pipelines.yml
Comment (Francisco-Gamino) [reply azure-pipelines.yml]: Fixed in the latest cherry-pick.

**Review by michaelpeng36 — APPROVED** [review]
LGTM

**Pull request merged**
